src/server/game/Entities/Player/Player.cpp

```diff
@@ -16440,12 +16441,16 @@ bool Player::HasQuestForItem(uint32 itemid) const
 
         QuestStatusData const& q_status = qs_itr->second;
 
-        if (q_status.Status == QUEST_STATUS_INCOMPLETE)
+        if (q_status.Status == QUEST_STATUS_INCOMPLETE || q_status.Status == QUEST_STATUS_COMPLETE)
         {
             Quest const* qinfo = sObjectMgr->GetQuestTemplate(questid);
             if (!qinfo)
                 continue;
 
+            if (questid == excludeQuestId)
+                excludeQuestId = 0;
+                return true;
```

Comment:
Yea, I get that. It just so happens that always returning true just happened to make it look like the patch was working as intended. As I explained to Shauren in IRC. A simple, "Hey, some braces here...." would have been enough. If I say I test something, then I test it.